hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/transactionalqueue/TransactionalQueueAddAllMessageTask.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.transactionalqueue;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.TransactionalQueueAddAllCodec;
+import com.hazelcast.client.impl.protocol.task.AbstractTransactionalMessageTask;
+import com.hazelcast.collection.impl.queue.QueueService;
+import com.hazelcast.instance.impl.Node;
+import com.hazelcast.internal.nio.Connection;
+import com.hazelcast.security.permission.ActionConstants;
+import com.hazelcast.security.permission.QueuePermission;
+import com.hazelcast.transaction.TransactionContext;
+import com.hazelcast.transaction.TransactionalQueue;
+
+import java.security.Permission;
+
+public class TransactionalQueueAddAllMessageTask
+        extends AbstractTransactionalMessageTask<TransactionalQueueAddAllCodec.RequestParameters> {
+
+    public TransactionalQueueAddAllMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected Object innerCall() throws Exception {
+        final TransactionContext context = endpoint.getTransactionContext(parameters.txnId);
+        final TransactionalQueue queue = context.getQueue(parameters.name);
+        return queue.addAll(parameters.items);
```

Comment:
how do we utilize `timeout`?

Reply:
Thank you for noticing this. I've addressed it.